src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py

```diff
@@ -21,13 +22,53 @@
     try_float_or_none,
     try_int,
     try_int_or_none,
+    try_str_from_series,
+    try_str_from_series_or_none,
 )
 
 BLOCKS_LINE_REGEX = r"^##BLOCKS=\s*(\d+)$"
 END_LINE_REGEX = "~End"
 EXPORT_VERSION = "1.3"
 
 
+def is_float(value: Any) -> bool:
+    try:
+        number = float(value)
+        return False if math.isnan(number) else True
+    except Exception:
+        return False
```

Comment:
catching `Exception` is generally a bad practice, as is making try blocks too large. How about instead:

```suggestion
    try:
        number = float(value)
    except ValueError:
        return False
    return not math.isnan(number)
```